docs/freedesktop-quick-reference.rst

```diff
@@ -37,25 +37,53 @@ command, *Type* and *Icon*:
 
     [Desktop Entry]
     Name=Gnome Dictionary
-    Exec=org.gnome.Dictionary
+    Exec=gnome-dictionary
     Type=Application
     Icon=org.gnome.Dictionary
 
 Your desktop file should be prefixed with your application's appid and
-placed in ``/app/share/applications/``, you should use
-``desktop-file-validate`` to check your file for errors before including
-it.
+placed in ``/app/share/applications/`` within your app.
 
 Example:
 
 ::
 
     /app/share/applications/org.gnome.Dictionary.desktop
 
-You can find more information
-`here <https://wiki.archlinux.org/index.php/desktop_entries>`__. If
-interested, you can read also the full spec
-`here <https://standards.freedesktop.org/desktop-entry-spec/latest/>`__.
+It's recommended to use ``desktop-file-validate`` to check your file
+for errors before including it.
+
+A special note about the ``Exec`` line: When installing the app on a user's
+system, Flatpak will automatically transform your desktop file so that the app
+will be started through Flatpak. For example, when installing
+``org.gnome.Dictionary``, a file will be created at
+``/var/lib/flatpak/exports/share/applications/org.gnome.Dictionary.desktop``.
+In this file, the ``Exec`` line looks like this::
+
+    Exec=/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=gnome-dictionary org.gnome.Dictionary
+
+The command from the original desktop file will be part of the
+``--command`` argument to Flatpak. This means that in most cases, it
+should match the value of the ``command:`` line in your app's manifest.
+
+If you want the ``--command`` argument to be omitted from the ``flatpak
+run`` command in the generated desktop file, you can just leave the
+``Exec`` value in the source desktop file empty::
+
+    [Desktop Entry]
+    Name=Gnome Dictionary
+    Exec=
```

Comment:
I like the simplicity of this solution (I already had to type `org.myorg.me.MyApp` in my repo over a dozen times, so omitting it here is relieving), but for me I get the following warning when I leave it empty:

```
(flatpak build-export:189863): GLib-CRITICAL **: 22:15:27.398: g_path_is_absolute: assertion 'file_name != NULL' failed
```

Not sure that we should suggest this as a "best practice" in the official docs, if the tools issue a warning about it.

Reply:
Yeah, better to change the tooling to explicitly support this first before recommending it, if it's a needed feature.

Reply:
https://github.com/flatpak/flatpak/pull/4817

Reply:
Thanks for making that change. However even if it's merged soon, people will still see the critical warning message until they're on a new enough version of Flatpak (which tends to take a long time). I don't know if it would also cause Flathub builds to fail, or if it would just add something scary to the logs. So we should either (a) split out this change into a separate MR and hold off on merging it for a while, or (b) mention that an empty Exec line should only be used with recent versions of Flatpak due to the warning message (or both).

Reply:
What about a note, indicating that flatpak versions <=$current may show a warning message and that this can be ignored? It's just a warning, it still works as intended with the currently released flatpak version.

(And this is only relevant for the person exporting a build to a repository, it's not shown during the flatpak build itself or when installing the flatpak.)

Reply:
Are you sure it works? Looking at the code, it seems like `convert_app_absolute_path()` would return a path ending in "bin/" when given a `NULL` path.

Reply:
I can confirm that it leaves the `Exec=` line empty after transformation (I'm looking at the correct file, `~/.local/share/flatpak/app/org.myorg.me.MyApp/current/active/files/share/applications/org.myorg.me.MyApp.desktop`, right?) and that the applications successfully starts when launched through my desktop's application launcher regardless.

Reply:
The file at `flatpak/app/org.myorg.me.MyApp/current/active/files/share/applications/org.myorg.me.MyApp.desktop` should contain the original empty `Exec=` line.

The file at `flatpak/exports/share/applications/org.myorg.me.MyApp.desktop` (this one should be processed by your app launcher) contains the transformed `Exec=/usr/bin/flatpak run --branch=master --arch=org.myorg.me.MyApp`.

Reply:
Oops, sorry then. Looking at the correct file now, when leaving the `Exec=` empty, it gets transformed to `Exec=/usr/bin/flatpak run --branch=master --arch=x86_64 org.myorg.me.MyApp` (which runs the applications successfully).